### DISCUSSION TITLE
Supress metadata printout when outputting to stdout (WSL)

### DISCUSSION THREAD

**Discussion by mbojan** [quarto-dev/quarto-cli#619] [Q&A] [answered]
I'm playing with using the quarto CLI and output to stdout. At this moment I can:

```
$ quarto render test.qmd -t plain -o -
pandoc -o /tmp/quarto-sessionf2c34e6c/e419b0c8/a187d20f.txt
  to: plain
  standalone: true
  default-image-extension: png
  filters:
    - crossref

metadata
  author: mbojan

mbojan

Some text.
```

Is there an option to supress printing the metadata (the `pandoc` and `metadata` "sections")? There is `--quiet` but it suppresses the document content too. Thanks!

PS. I'm test-driving couple of things and plan to post some more questions. I'm assuming Discussions are better than Issues for that. Let me know if not.

**Edit**:

I'm running

```
$ quarto -v
0.9.293
```

and WSL:
![obraz](https://user-images.githubusercontent.com/636574/165768942-054a89d5-6b99-4227-bd85-e1d91c1429fd.png)

**Comment by cscheid** [comment]
I think the easiest solution for now is to use `mktemp` on a shell, wait for `quarto` to finish, and then `cat` that file into the rest of your pipeline. I also just tried named pipes but they break quarto for a different reason. See https://github.com/quarto-dev/quarto-cli/issues/635 for details, but I think you're likely to run into sharp edges here, and using a temporary filename is very unlikely to be surprising to quarto's internal code.

**Comment by jjallaire** [comment]
The printing of the metadata should be to stderr rather than stdout, could you just redirect stderr? `--quiet` definitely shouldn't suppress stdout. I'll take a look at that soon.

**Comment by jjallaire** [comment, marked as answer]
I tested this and it does prevent printing of the extra output:

```bash
quarto render test.qmd -t plain -o - 2>/dev/null
```

Further, this also dropped the extra output and did indeed write the document to stdout:

```bash
quarto render test.qmd -t plain -o - --quiet
```

**Reply by mbojan** [reply]
Thanks @jjallaire , but neither of those two commands work as intended on WSL:

![obraz](https://user-images.githubusercontent.com/636574/165767262-b50f232b-7c7f-4e65-92b1-cf8b62f25868.png)

The result of the first one (redirecting stderr) is identical - nothing.

**Reply by mbojan** [reply]
I have updated issue description that it seems the problem is on WSL specifically.

**Reply by jjallaire** [reply]
If the basic stdout and stderr behavior isn't working as expected on WSL that might be a tough one for us to resolve. I just tried getting WSL installed on my Windows VM but it failed both times I tried it :-(

@cderv Do you have a working WSL instance and if so can you repro this? It would be also interesting to know if just a plain pandoc call exhibits the same behavior under WSL.

**Reply by mbojan** [reply]
> It would be also interesting to know if just a plain pandoc call exhibits the same behavior under WSL.

It seems to be working on my WSL:

```
$ pandoc -i file.md -t plain --quiet -o -


TITLE


Content.
```

**Reply by jjallaire** [reply]
what about this:

```bash
$ quarto pandoc -i file.md -t plain --quiet -o -
```

**Reply by cderv** [reply]
> Do you have a working WSL instance and if so can you repro this?

Yes I have a working WSL - I do most of my Linux work in it. 

Using latest quarto, I cannot reproduce. It works for me doing this: 

````sh
$ quarto render test.qmd -t plain --quiet -o -
Demo


Heading

Content
````

Not setting quiet will show the all log

````sh
$ quarto render test.qmd -t plain -o -
pandoc -o /tmp/quarto-sessionc45d6445/82d45df3/85ce81f7.txt
  to: plain
  standalone: true
  default-image-extension: png
  filters:
    - crossref
  
metadata
  title: Demo
  
Demo


Heading

Content
````

Regarding using Pandoc directly 

````sh
# No argument ------
$ pandoc -i test.qmd -t plain
[WARNING] Could not deduce format from file extension .qmd
  Defaulting to markdown
Heading

Content

## reverse output with quarto - warning after 
$ quarto pandoc -i test.qmd -t plain
Heading

Content
[WARNING] Could not deduce format from file extension .qmd
  Defaulting to markdown

# Using quiet -------
$ pandoc -i test.qmd -t plain --quiet
Heading

Content

## However, no  output with quarto
$ quarto pandoc -i test.qmd -t plain --quiet


# Adding -o
$ pandoc -i test.qmd -t plain --quiet -o -
Heading

Content

## However, no  output with quarto
$ quarto pandoc -i test.qmd -t plain --quiet -o -
````


So in conclusion 
* using `--quiet` with `-o -` seems to work as expected with `quarto render`. 
* using `--quiet` with `quarto pandoc` will hide all output, but it should only hide the Warning like with `pandoc`

<details><summary>simple qmd file</summary>

````markdown
---
title: "Demo"
format: html
---

# Heading

Content
````

</details>

**Reply by mbojan** [reply]
Interesting... I'll try again after updating to the most recent version of Quarto. 

@cderv do you have WSL version 1 or 2?

**Reply by cderv** [reply]
```
❯ wsl --list --verbose
  NAME                   STATE           VERSION
* Ubuntu-20.04           Running         2
```

version 2
